core/services/keeper/upkeep_executer_test.go

```diff
@@ -279,41 +299,22 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
 
 		registryMock := cltest.NewContractMockReceiver(t, ethMock, keeper.Registry1_1ABI, registry.ContractAddress.Address())
 		registryMock.MockResponse("checkUpkeep", checkUpkeepResponse)
-
+		registryMock.MockMatchedResponse(
+			"performUpkeep",
+			func(callArgs ethereum.CallMsg) bool { return true },
+			checkPerformResponse,
+		)
 		// turn falls somewhere between 20-39 (blockCountPerTurn=20)
 		// heads 20 thru 35 were skipped (e.g. due to node reboot)
 		head := cltest.Head(36)
 
 		executer.OnNewLongestChain(context.Background(), head)
-		runs := cltest.WaitForPipelineComplete(t, 0, job.ID, 1, 5, jpv2.Jrm, time.Second, 100*time.Millisecond)
+		runs := cltest.WaitForPipelineComplete(t, 0, job.ID, 1, 8, jpv2.Jrm, time.Second, 100*time.Millisecond)
 		require.Len(t, runs, 1)
 		assert.False(t, runs[0].HasErrors())
 		etxs[0].AwaitOrFail(t)
 		waitLastRunHeight(t, db, upkeep, 36)
 
-		// heads 37, 38 etc do nothing
-		for i := 37; i < 40; i++ {
-			head = cltest.Head(i)
-			executer.OnNewLongestChain(context.Background(), head)
-		}
-
-		// head 40 triggers a new run
-		head = cltest.Head(40)
-
-		txm.On("CreateEthTransaction",
-			mock.MatchedBy(func(newTx txmgr.NewTx) bool { return newTx.GasLimit == gasLimit }),
-		).
-			Once().
-			Return(txmgr.EthTx{}, nil).
-			Run(func(mock.Arguments) { etxs[1].ItHappened() })
-
-		executer.OnNewLongestChain(context.Background(), head)
-		runs = cltest.WaitForPipelineComplete(t, 0, job.ID, 2, 5, jpv2.Jrm, time.Second, 100*time.Millisecond)
-		require.Len(t, runs, 2)
-		assert.False(t, runs[1].HasErrors())
-		etxs[1].AwaitOrFail(t)
-		waitLastRunHeight(t, db, upkeep, 40)
-
```

Comment:
not sure why we are not testing this

Reply:
this test description was about skipping blocks which is still tested in the first half. but the second half was testing the logic of the old turn taking and didn't really make sense with the new turn taking. It also relied on the grace period being 0 to force an unrealistic single keeper scenario. This lead it to became flaky because it was really only passing due to its speedy looping of blocks.